distributed/tests/test_nanny.py

```diff
@@ -145,6 +145,31 @@ def test_close_on_disconnect(s, w):
         assert time() < start + 9
 
 
+class Something(Worker):
+    # a subclass of Worker which is not Worker
+    pass
+
+
+class Nanny2(Nanny):
+    # a subclass with alternate Worker class to create
+    Worker = Something
+
+
+@gen_cluster(client=True, ncores=[])
+def test_nanny_worker_class(c, s):
+    w = Nanny(s.ip, s.port, ncores=1, loop=s.loop)
```

Comment:
You'll want to add `Worker=` to the constructor here.  

Alternatively, you might consider using gen_cluster for this with 

    gen_cluster(client=True, Worker=Nanny, worker_kwargs={'Worker': Something})

This is a bit nicer because it will clean up things slightly more reliably.

Reply:
ah ,sorry